core_lib/src/structure/object.h

```diff
@@ -82,10 +82,14 @@ class Object : public QObject
     QString copyFileToDataFolder( QString strFilePath );
 
     // Color palette
-    ColourRef getColour( int i ) const;
+    ColourRef getColour( int index ) const;
+    void useAsTempPaletteColor(QColor color) { mFrontColor = ColourRef(color, "Front Color"); }
```

Comment:
Could you please explain to me why adding `useAsTempPaletteColor()`?

Reply:
Because since vector is dependant on mPalette, if there are no swatches left, the color will become black. This makes sure that it'll use the front color instead if that's the case.
Are you against the method, should I add it to Object::setColour instead?

The other solution is to not allow all swatches to be deleted but I find this to be a workaround rather than a proper solution.

Reply:
Probably the best solution for now (I mean before you implement the connected color), we probably couldn't allow users to delete all palette colors, need to at least keep one color.

Reply:
Okay so i'll remove the useTemp solution and make it impossible to delete the last remaining swatch.

Reply:
Thanks :)